salt/modules/x509_v2.py

```diff
@@ -0,0 +1,1933 @@
+
+
+def _generate_pk(algo="rsa", keysize=None):
+    if algo == "rsa":
+        return x509util.generate_rsa_privkey(keysize=keysize or 2048)
+    if algo == "ec":
+        return x509util.generate_ec_privkey(keysize=keysize or 256)
+    if algo == "ed25519":
+        return x509util.generate_ed25519_privkey()
+    if algo == "ed448":
+        return x509util.generate_ed448_privkey()
+    raise SaltInvocationError(
+        f"Invalid algorithm specified for generating private key: {algo}. Valid: "
+        "rsa, ec, ed25519, ed448"
+    )
+
+
+def _get_signing_policy(name):
+    if name is None:
+        return {}
+    policies = __salt__["pillar.get"]("x509_signing_policies", {}).get(name)
+    policies = policies or __salt__["config.get"]("x509_signing_policies", {}).get(name)
+    if isinstance(policies, list):
+        dict_ = {}
+        for item in policies:
+            dict_.update(item)
+        policies = dict_
+    return policies or {}
+
+
+def _parse_dn(subject):
+    """
+    Returns a dict containing all values in an X509 Subject
+    """
+    ret = OrderedDict()
+    for nid_name, oid in x509util.NAME_OID.items():
+        try:
+            ret[nid_name] = subject.get_attributes_for_oid(oid)[0].value
+        except IndexError:
+            continue
+    return ret
+
+
+def _get_name_hash(name, digest="sha1"):
+    """
+    Returns the OpenSSL name hash.
+    This is the first four bytes of the SHA1 (pre v1: MD5) hash
+    of the DER-encoded form of name. On little-endian systems,
+    OpenSSL inverts the bytes.
+    """
+    if digest.lower() not in ["sha1", "md5"]:
+        raise ValueError(
+            f"Invalid hashing algorithm for name hash: {digest}. "
+            "Only SHA1 and MD5 are allowed"
+        )
+    hsh = hashes.Hash(x509util.get_hashing_algorithm(digest))
+    hsh.update(name.public_bytes())
+    res = hsh.finalize()[:4]
+    if sys.byteorder == "little":
+        res = res[::-1]
+    return res
+
+
+def _parse_extensions(extensions):
+    ret = {}
+    for extname, _, oid in x509util.EXTNAMES:
+        try:
+            ext = extensions.get_extension_for_oid(oid)
+        except cx509.ExtensionNotFound:
+            continue
+        ret[extname] = x509util.render_extension(ext)
+    return ret
+
+
+def _parse_crl_entry_extensions(extensions):
+    ret = {}
+    for extname, _, oid in x509util.EXTNAMES_CRL_ENTRY:
+        try:
+            ext = extensions.get_extension_for_oid(oid)
+        except cx509.ExtensionNotFound:
+            continue
+        ret[extname] = x509util.render_extension(ext)
+    return ret
+
+
+def _match_minions(test, minion):
+    if "@" in test:
+        # This essentially asks the minion if it is allowed to receive
+        # certificates with the signing policy. Implementing a match runner
+        # would plug that security hole somewhat, and fully if only pillars
+        # are used.
+        match = __salt__["publish.publish"](tgt=minion, fun="match.compound", arg=test)
```

Comment:
This was inherited from the old module and is not a good practice. I would honestly prefer to use https://github.com/saltstack/salt/pull/63297 instead right from the start. Not a blocker ofc since it's opt-in.